packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1819,6 +1807,23 @@ class TextSelectionGestureDetectorBuilder {
         && selection.end >= textPosition.offset;
   }
 
+  /// Returns true if position was on selection.
+  bool _positionOnSelection(Offset position, TextSelection? targetSelection) {
+    if (targetSelection == null) {
+      return false;
+    }
+
+    final TextPosition textPosition = renderEditable.getPositionForPoint(position);
```

Comment:
I think all should be fine, but I can take a look and see if i can make any improvements. In reality we don't need to call this method on every drag update, since it is always using the same `dragStartGlobalPosition` and `_dragStartSelection` that was tracked at the beginning of the drag in `onStart`. We should be able to call this once, and then it should be true for the remaining drag update events.

Reply:
Fixed this by using a new flag `dragBeganOnPreviousSelection`, it is set on drag update only if it is null by calling `_positionOnSelection` which calls `getPositionForPoint` in question.  Here it is called once for the entirety of the drag. It is reset to null on drag end.